ff/src/fields/models/quadratic_extension.rs

```diff
@@ -192,15 +193,14 @@ impl<P: QuadExtParameters> One for QuadExtField<P> {
 
 impl<P: QuadExtParameters> Field for QuadExtField<P> {
     type BasePrimeField = P::BasePrimeField;
-    type BasePrimeFieldArr = Vec<Self::BasePrimeField>;
+    type BasePrimeFieldIter = Box<dyn Iterator<Item = Self::BasePrimeField>>; 
```

Comment:
@Pratyush, I had to box this, otherwise it refuses to compile with an iterator whose size is not known at compile time with: `the size for values of type (dyn Iterator<Item = Self::BasePrimeField>) cannot be known at compilation time` Error. Perhaps because `Field` is `Sized`. I'm not sure if you had another work around for this in mind.

Reply:
Ah! now I see you had already mentioned the issue and suggested `Chain`. I'll try to replace  with Chain then. Sorry for premature question.